doc/index.rst

```diff
@@ -89,6 +89,7 @@ Catalyst
    demos/magic_distillation_demo
    demos/adaptive_circuits_demo
    demos/qml/tutorial_qubit_rotation
+   QML Optimization <demos/qml_optimization>
```

Comment:
As this demo is using `Optax` for optimizing the workflow, would you like to add this to the index title
 
```suggestion
   QML Optimization with Optax <demos/qml_optimization>
```